ui-integration/src/App.js

```diff
@@ -0,0 +1,298 @@
+          diagnosisColumns[columnName][levelKey] = options;
+        });
+      }
+    });
+    setDiagnosisOptions(diagnosisColumns);
+  };
+
+  const filterJsonData = () => {
+    if (!responseData || Object.keys(selectedDiagnosis).length === 0) return;
+  
+    const filtered = { ...responseData };
+  
+    Object.keys(selectedDiagnosis).forEach((columnName) => {
+      const selectedColumn = selectedDiagnosis[columnName];
+      if (!filtered[columnName]?.Annotations?.Levels || !selectedColumn) return;
+  
+      const levels = filtered[columnName].Annotations.Levels;
+  
+      Object.keys(selectedColumn).forEach((levelKey) => {
+        const selectedValue = selectedColumn[levelKey];
+        if (!levels[levelKey]) return;
+  
+        if (selectedValue === "no-match") {
+          delete levels[levelKey]; // Remove the key entirely
+        } else if (selectedValue) {
+          levels[levelKey] = {
+            [selectedValue]: levels[levelKey][selectedValue],
+          };
+  
+          if (filtered[columnName].Levels && Array.isArray(filtered[columnName].Levels[levelKey])) {
+            filtered[columnName].Levels[levelKey] = filtered[columnName].Levels[levelKey].filter((term) =>
+              levels[levelKey][selectedValue]?.Label === term
+            );
+          }
+        }
+  
+        // Flatten the structure if only one key exists
+        if (levels[levelKey] && Object.keys(levels[levelKey]).length === 1) {
+          levels[levelKey] = levels[levelKey][Object.keys(levels[levelKey])[0]];
+        }
+      });
+    });
+  
+    // Replace numeric Levels field both inside and outside the Annotations part with "units": "Arbitrary units"
+    Object.keys(filtered).forEach((columnName) => {
+      const levels = filtered[columnName]?.Levels;
+      const annotationLevels = filtered[columnName]?.Annotations?.Levels;
+  
+      // Replace numeric Levels field outside the Annotations part
+      if (levels && Object.keys(levels).every(levelKey => /^\d+$/.test(levelKey))) {
+        delete filtered[columnName].Levels;
+        filtered[columnName].units = "Arbitrary units";
+      }
+  
+      // Replace numeric Levels field inside the Annotations part
+      if (annotationLevels) {
+        Object.keys(annotationLevels).forEach((levelKey) => {
+          if (/^\d+$/.test(levelKey)) {  // Check if the key is numeric
+            delete annotationLevels[levelKey];
+            filtered[columnName].Annotations["units"] = "Arbitrary units";
+          }
+        });
+      }
+    });
+  
+    setFilteredData(filtered);
+  };
+  
+  
+  
+
+  const downloadJson = () => {
+    if (!filteredData) return;
+
+    const blob = new Blob([JSON.stringify(filteredData, null, 2)], { type: "application/json" });
+    const url = window.URL.createObjectURL(blob);
+    const a = document.createElement("a");
+    a.href = url;
+    a.download = "filtered_data.json";
+    document.body.appendChild(a);
+    a.click();
+    a.remove();
+  };
```

Comment:
I think that could be simplified by using the file saver

```suggestion
const downloadJson = () => {
  if (!filteredData) return;

  const blob = new Blob([JSON.stringify(filteredData, null, 2)], { type: "application/json" });
  saveAs(blob, "filtered_data.json");
};
```